lib/semaphore.rb

```diff
@@ -47,14 +47,16 @@ def exclusively(lock_key, expiration: nil, &block)
 
     def locked?(lock_key)
       token = nil
-      perform_action_on_semaphore do
-        # Because of https://github.com/dv/redis-semaphore/issues/40, we need a
-        # complex check if a lock is set. After redis-semaphore#40 is fixed,
-        # this might be replaced by the use of my_redis_semaphore.locked?
-        sema = retrieve_semaphore(lock_key, expiration: LOCK_ACTION_EXPIRATION)
-        # lock(0) returns `"0"` if it was free and `false` if it was locked
-        token = sema.lock(0)
-        sema.unlock if token
+      perform_action_on_semaphore(:lock) do
+        perform_action_on_semaphore(:unlock) do
+          # Because of https://github.com/dv/redis-semaphore/issues/40, we need a
```

Comment:
I don't care at this point. See the opening description and the referenced issue.